lib/spack/spack/test/database.py

```diff
@@ -470,6 +473,27 @@ def test_015_write_and_read(mutable_database):
         assert new_rec.installed == rec.installed
 
 
+def test_016_read_multiple_db(tmpdir, mock_fetch, mock_store, install_mockery):
+    '''
+    Because spec.prefix reads from the database to check for existing specs,
+    we need to store the prefix from the database at read time in case the spec
+    comes from a non-default database (like the bootstrap store)
+    '''
+    with spack.store.use_store(str(tmpdir)):
+        with spack.store.db.write_transaction():
+            orig_prefix = spack.spec.Spec('libelf').concretized().prefix
+            install('libelf')
```

Comment:
I think this might solve the issue with MacOS:
```suggestion
        orig_prefix = spack.spec.Spec('libelf').concretized().prefix
        install('libelf')
```
I didn't try out this solution, but I suspect it may be related to the fact that on linux we fork while on Mac we do not (the test fails only on Mac waiting for a read lock to be available).

Reply:
I think this failure is related to something I've fixed in another PR. I'll try rebasing this once #22588 is in.